test/integration/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,934 @@
+    // The zone could not be updated due to zone is not found.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'Address',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId: invalidID,
+      ifMatch: zoneEtag,
+      name: 'an example of zone not found',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of zone not found',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      status: 400,
+    });
+  });
+
+  test('listAvailableServicerefTargets() - 200', async () => {
+    const params = {
+      type: 'all',
+      transactionId: uuidv4(),
+    };
+
+    const res = await contextBasedRestrictionsService.listAvailableServicerefTargets(params);
+    expect(res).toBeDefined();
+    expect(res.status).toBe(200);
+    expect(res.result).toBeDefined();
+  });
+
+  test('listAvailableServicerefTargets() - 400 - invalid type', async () => {
+    // type value is not supported
+    const params = {
+      type: 'invalid-type',
+      transactionId: uuidv4(),
+    };
+
+    await expect(
+      contextBasedRestrictionsService.listAvailableServicerefTargets(params)
+    ).rejects.toMatchObject({
+      status: 400,
+    });
+  });
+
+  test('createRule() - 201', async (done) => {
+    // Request models needed by this operation.
+
+    // RuleContextAttribute
+    const ruleContextAttributeModel = {
+      name: 'networkZoneId',
+      value: zoneId,
+    };
+
+    // RuleContext
+    const ruleContextModel = {
+      attributes: [ruleContextAttributeModel],
+    };
+
+    // ResourceAttribute
+    const resourceAttributeAccountIdModel = {
+      name: 'accountId',
+      value: accountId,
+      operator: 'stringEquals',
+    };
+
+    const resourceAttributeServiceNameModel = {
+      name: 'serviceName',
+      value: serviceName,
+      operator: 'stringEquals',
+    };
+
+    // ResourceTagAttribute
+    const resourceTagAttributeModel = {
+      name: 'aTagName',
+      value: 'aTagValue',
+      operator: 'stringEquals',
+    };
+
+    // Resource
+    const resourceModel = {
+      attributes: [resourceAttributeAccountIdModel, resourceAttributeServiceNameModel],
+      tags: [resourceTagAttributeModel],
+    };
+
+    const params = {
+      contexts: [ruleContextModel],
+      resources: [resourceModel],
+      description: 'this is an example of rule',
+      transactionId: uuidv4(),
+    };
+
+    let res;
+    try {
+      res = await contextBasedRestrictionsService.createRule(params);
+    } catch (err) {
+      done(err);
+    }
+
+    expect(res).toBeDefined();
+    expect(res.status).toBe(201);
+    expect(res.result).toBeDefined();
+
+    ruleId = res.result.id;
+    ruleEtag = res.headers.etag;
+
+    done();
+  });
+
+  test('createRule() - 400 - missing accountId', async () => {
```

Comment:
It is the same as previously, this test and the following one exercise `400`. It is up to you if this one remains here.

Reply:
removed test cases "createRule() - 400 - missing accountId" and "createRule() - 400 - with a service not cbr enabled"